include/seahorn/PredicateAbstraction.hh

```diff
@@ -0,0 +1,151 @@
+#ifndef PREDICATE_ABSTRACTION__HH_
+#define PREDICATE_ABSTRACTION__HH_
+
+#include "llvm/Pass.h"
+#include "llvm/IR/Module.h"
+#include "seahorn/HornClauseDB.hh"
+#include "seahorn/HornifyModule.hh"
+
+#include "ufo/Expr.hpp"
+#include "ufo/Smt/Z3n.hpp"
+#include "ufo/Smt/EZ3.hh"
+#include "seahorn/HornClauseDBWto.hh"
+
+namespace seahorn
+{
+	using namespace llvm;
+
+	class HornDbModel
+	{
+	private:
+		ExprMap relToDefMap;
+		std::map<Expr, ExprVector> currentCandidates;
+	public:
+		HornDbModel() {}
+		void initModelFromFP(HornClauseDB &db, ZFixedPoint<EZ3> &fp);
+		void addDef(Expr fapp, Expr def);
```

Comment:
add comments for how the function should be used. Applies to other functions in this class as well.